packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts

```diff
@@ -0,0 +1,252 @@
+import * as cdk from '@aws-cdk/core';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as iam from '@aws-cdk/aws-iam';
+import * as path from 'path';
+import { prepareApp } from '@aws-cdk/core/lib/private/prepare-app';
+import { AmplifyRootStackResourceProps, AmplifyRootStackTemplateProps } from './types';
+import { JSONUtilities, pathManager } from 'amplify-cli-core';
+import { Template } from 'cloudform-types';
+
+const CFN_TEMPLATE_FORMAT_VERSION = '2010-09-09';
+const ROOT_CFN_DESCRIPTION = 'Root Stack for AWS Amplify CLI';
+
+export class AmplifyRootStack extends cdk.Stack {
+  public templateObj: AmplifyRootStackTemplateProps;
+  constructor(scope: cdk.Construct, id: string) {
+    super(scope, id);
+    this.templateOptions.templateFormatVersion = CFN_TEMPLATE_FORMAT_VERSION;
+    this.templateOptions.description = ROOT_CFN_DESCRIPTION;
+    const CfnParameters: { [k: string]: cdk.CfnParameter } = {};
+
+    CfnParameters.deploymentBucketName = new cdk.CfnParameter(this, 'DeploymentBucketName', {
+      type: 'String',
+      description: 'Name of the common deployment bucket provided by the parent stack',
+      default: 'DeploymentBucket',
+    });
+
+    CfnParameters.authRoleName = new cdk.CfnParameter(this, 'AuthRoleName', {
+      type: 'String',
+      default: 'AuthRoleName',
+    });
+
+    CfnParameters.unauthRoleName = new cdk.CfnParameter(this, 'UnauthRoleName', {
+      type: 'String',
+      default: 'UnauthRoleName',
+    });
+    // Reources
+
+    let CfnResources: AmplifyRootStackResourceProps = {};
+    CfnResources.DeploymentBucket = new s3.CfnBucket(this, 'DeploymentBucket', {
+      bucketName: CfnParameters['deploymentBucketName'].valueAsString,
+    });
+
+    CfnResources.DeploymentBucket.applyRemovalPolicy(cdk.RemovalPolicy.RETAIN);
+
+    CfnResources.authRole = new iam.CfnRole(this, 'AuthRole', {
+      roleName: CfnParameters['authRoleName'].valueAsString,
+      assumeRolePolicyDocument: {
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Sid: '',
+            Effect: 'Deny',
+            Principal: {
+              Federated: 'cognito-identity.amazonaws.com',
+            },
+            Action: 'sts:AssumeRoleWithWebIdentity',
+          },
+        ],
+      },
+    });
+
+    CfnResources.unauthRole = new iam.CfnRole(this, 'UnauthRole', {
```

Comment:
We also add a custom lambda resource in the root-stack when you run `amplify add auth` and push. Is that accounted for or will that happen dynamically and cannot be overriden?

Reply:
This one ill take over when we start overriding of Auth Category